graphics/src/AssimpLoader.cc

```diff
@@ -0,0 +1,704 @@
+  /// a metalness and roughness map
+  /// \param[in] _img the image to split
+  /// \return a pair of image pointers with the first being the metalness
+  /// map and the second being the roughness map
+  public: std::pair<ImagePtr, ImagePtr>
+          SplitMetallicRoughnessMap(const common::Image& _img) const;
+
+  /// \brief Convert an assimp mesh into a gz::common::SubMesh
+  /// \param[in] _assimpMesh the assimp mesh to load
+  /// \param[in] _transform the node transform for the mesh
+  /// \return the converted common::Submesh
+  public: SubMesh CreateSubMesh(const aiMesh* _assimpMesh,
+                                const math::Matrix4d& _transform);
+
+  /// \brief Recursively create submeshes scene starting from the root node
+  /// \param[in] _scene the assimp scene
+  /// \param[in] _node the node being processed
+  /// \param[in] _node the transform of the node being processed
+  /// \param[out] _mesh the common::Mesh to edit
+  public: void RecursiveCreate(const aiScene* _scene,
+                               const aiNode* _node,
+                               const math::Matrix4d& _transform,
+                               Mesh* _mesh);
+
+  /// \brief Recursively create the skeleton starting from the root node
+  /// \param[in] _node the node being processed
+  /// \param[in] _parent the parent skeleton node
+  /// \param[in] _transform the transform of the current node
+  /// \param[in] _boneNames set of bone names, used to skip nodes without a bone
+  public: void RecursiveSkeletonCreate(
+          const aiNode* _node,
+          SkeletonNode* _parent,
+          const math::Matrix4d& _transform,
+          const std::unordered_set<std::string> &_boneNames);
+
+  /// \brief Recursively store the bone names starting from the root node
+  /// to make sure that only nodes that map to a bone are added to the skeleton
+  /// \param[in] _scene the assimp scene
+  /// \param[in] _node the node being processed
+  /// \param[out] _boneNames set of bone names populated while recursing
+  public: void RecursiveStoreBoneNames(
+          const aiScene *_scene,
+          const aiNode* _node,
+          std::unordered_set<std::string>& _boneNames);
+
+  /// \brief Apply the the inv bind transform to the skeleton pose.
+  /// \remarks have to set the model transforms starting from the root in
+  /// breadth first order. Because setting the model transform also updates
+  /// the transform based on the parent's inv model transform. Setting the
+  /// child before the parent results in the child's transform being
+  /// calculated from the "old" parent model transform.
+  /// \param[in] _skeleton the skeleton to work on
+  public: void ApplyInvBindTransform(SkeletonPtr _skeleton);
+};
+
+// Utility function to convert to std::string from aiString
+static std::string ToString(const aiString& str)
+{
+  return std::string(str.C_Str());
+}
+
+math::Color AssimpLoader::Implementation::ConvertColor(aiColor4D& _color)
```

Comment:
Add this in all method/function implementations
```suggestion
/////////////////////////////////////////////////
math::Color AssimpLoader::Implementation::ConvertColor(aiColor4D& _color)
```

Reply:
https://github.com/gazebosim/gz-common/pull/393/commits/d647967caf54768f45b9e9bd296c858c85ec4487